src/resmom/mom_comm.c

```diff
@@ -1883,6 +1883,197 @@ static	char	bail_format[] = "dis read failed: %s";
 
 /**
  * @brief
+ *	Send resources_used values to the MS via
+ *	'stream' descriptor.
+ *
+ * @param[in] stream - descriptor pathway to MS.
+ * @param[in] jobid - the jobid of the owning job.
+ *
+ * @return  error code
+ * @retval -1     error
+ * @retval  0     Success
+ *
+ */
+int
+send_resc_used_to_ms(int stream, char *jobid)
+{
+	attribute		*at;
+	attribute_def		*ad;
+	svrattrl		*pal;
+	svrattrl		*nxpal;
+	svrattrl		*sister;
+	pbs_list_head		lhead;
+	pbs_list_head		send_head;
+	extern	int		resc_access_perm;
+	svrattrl		 *psatl;
+	job			*pjob;
+	int			ret;
+
+	if (jobid == NULL)
+		return (-1);
+
+	if (stream == -1)
+		return (-1);
+
+	pjob = find_job(jobid);
+	if (pjob == NULL)
+		return (-1);
+
+	at = &pjob->ji_wattr[(int)JOB_ATR_resc_used];
+	if (at->at_type != ATR_TYPE_RESC) {
+		return (-1);
+	}
+	ad = &job_attr_def[(int)JOB_ATR_resc_used];
+	resc_access_perm = ATR_DFLAG_MGRD;
+
+	memset(&lhead, 0, sizeof(lhead));
+	CLEAR_HEAD(lhead);
+
+	(void)ad->at_encode(at,
+		&lhead, ad->at_name,
+		NULL, ATR_ENCODE_CLIENT, NULL);
+
+	memset(&send_head, 0, sizeof(send_head));
+	CLEAR_HEAD(send_head);
+
+	pal = (svrattrl *)GET_NEXT(lhead);
+	psatl = pal;
+
+	while (pal != (svrattrl *)0) {
+		nxpal = (struct svrattrl *)GET_NEXT(pal->al_link);
+
+		/* no need to track the resources automatically sent to MS */
+		/* like 'cput', 'mem', and 'cpupercent',but only those */
+		/* resources that are set in a mom hook */
+		if (((pal->al_flags & ATR_VFLAG_HOOK) != 0) && \
+		     (strcmp(pal->al_resc, "cput") != 0) && \
+		     (strcmp(pal->al_resc, "mem") != 0) && \
+		     (strcmp(pal->al_resc, "cpupercent") != 0)) {
```

Comment:
No need to add a similar test, for send_resc_used_to_ms() is actually sending resources_used values that have been set in encode_used().